Core/src/EventData/CorrectedTransformationFreeToBound.cpp

```diff
@@ -0,0 +1,132 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021-2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/EventData/detail/CorrectedTransformationFreeToBound.hpp"
+
+#include "Acts/Surfaces/Surface.hpp"
+
+/// Get the non-linearity corrected bound parameters and its covariance
+std::optional<std::tuple<Acts::BoundVector, Acts::BoundSymMatrix>>
+Acts::detail::CorrectedFreeToBoundTransformer::operator()(
+    const Acts::FreeVector& freeParams,
+    const Acts::FreeSymMatrix& freeCovariance, const Acts::Surface& surface,
+    const Acts::GeometryContext& geoContext, NavigationDirection navDir) {
+  // Get the incidence angle
+  Vector3 dir = freeParams.segment<3>(eFreeDir0);
+  Vector3 normal = surface.normal(geoContext);
+  ActsScalar absCosIncidenceAng = std::abs(dir.dot(normal));
+  // No correction if the incidentAngle is small enough (not necessary ) or too
+  // large (correction could be invalid).
+  if (absCosIncidenceAng < cosIncidentAngleMinCutoff or
+      absCosIncidenceAng > cosIncidentAngleMaxCutoff) {
+    return std::nullopt;
```

Comment:
I guess the `nullopt` means to fall back to the normal freeToBound transformation?

Reply:
Yes, it meas the normal freeToBound transformation will be used. Comment has been added.